sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CollectionExpressionsSuite.scala

```diff
@@ -137,14 +137,32 @@ class CollectionExpressionsSuite
 
   test("ArrayBinarySearch") {
     // primitive type: boolean、byte、short、int、long、float、double
+    // boolean
```

Comment:
do we have non-foldable test cases in it?

Reply:
Yes, as follows